### DISCUSSION TITLE
Real Estate Tracking

### DISCUSSION THREAD

**Discussion by jakeau** [ghostfolio/ghostfolio#2851] [Ideas]
It would be cool to have the ability to track both mortgages and real-estate together in more depth, along with the change in prices, rent yield, etc.

**Comment by bastienjalbert** [comment]
+1
I agree

**Comment by joestump** [comment]
Came here to throw in my support for this feature. I looked into this a bit. There's apparently a [Zillow API](https://www.zillowgroup.com/developers/api/zestimate/zestimates-api/). I found [this post of 10 real estate APIs](https://www.softkraft.co/real-estate-apis/). Not surprisingly, they range in price from a few cents per API call to thousands per month. 

Honestly, as someone with multiple investment properties, I would 100% pay for an API the same way I pay for SimpleFIN for Actual Budget. 

Barring that, I can think of two other ways to approach this feature in the app:

* Allow me to toggle "include valuables" in my net worth calculations
* Create a new account type that is of type "Property" and I can then go in and add activities for adjustments in prices, capital investments (e.g. renovations), etc. 

That last approach would work really well for me as a rental/Airbnb owner. I could go in monthly/quarterly and add my income/expenses per property.

**Comment by pimterry** [comment]
Tracking house prices is quite subjective, but tracking mortgages should be doable very concretely with no external sources required. It should be possible to enter the initial total, interest rate & monthly payment (probably need to be a series of date-range values: 2% + $1000/month during 2023-1-1 to 2024-12-31, 1.8% + $900/month from 2025-1-1), and calculate the remaining value and how long it'll take to pay off.

Personally, I would want to manually enter the house price (and occasionally update it from a valuation or a best guess or Zillow or whatever) but automatically track the mortgage value each month. Given that, Ghostfolio should be able to include the net value in my totals.

The same would work for any other loans with interest too (student loans and all sorts).

**Reply by jakeau** [reply]
I've ended up moving from ghostfolio to my own simple python script to do this. 

Essentially what I'm doing: 
- My bank statements are emailed to me 
- PDF is parsed to pull out interest, rent repayments, mortgage repayments stored into POSTGRESQL
- I scrape an Australian real estate service similar to Zillow to track the estimated value, every month.


Not having touched ghostfolio in a year I don't think it could support something like this. 


Does Ghostfolio track different assets differently, if so could someone let me know and I'll see if I can implement something similar?

**Comment by ccoles146** [comment]
I just started using Ghostfolio but I think I've managed to do it.  Now tracking a generic house-price value index is quite hit and miss. I was able to upload the UK housing index and also a regional housing index but these still do not track very accurately.
However, what I did was:

1. Add Market Data Manually
2. Give it a name like GF_Property_UK
3. Edit the newly created Symbol and change the Currency and Asset Class as appropriate.
4. Go to Historical Market Data and manually import data points, either as CSV or simply once per year should be sufficient.

Then having created the "Value Index", you then need to purchase a quantity of these.

1. Create an Account called for example Real Estate GBP
2. Then manually add an activity, 
3. select your Real Estate account
4. Input GF_Property_UK
5. Qty. 1
6. Unit price is your purchase price.

So that's the property equity tracking.  Now if you want a net worth calculation then you'll need your mortgage.  I followed essentially the same principle. I didn't worry about calculating interest.  I get a yearly statement from my Mortgage provider.

1. Add Market Data Manually
2. Give it a name like GF_Mortgage_Property1
3. Edit the newly created Symbol and set the asset class to Liquidity
4. Go to Historical Market Data and manually import data points. This time the value should decrease over time. I create a value of the mortgage on 31-Dec each year.
5. Then at the same time that I buy a property, I sell a mortgage. (sort of works like a future).

I think that works out ok.  Let me know if you have any comments on this workflow.
